### PR TITLE
Core refactoring

### PR DESCRIPTION
**Pull request by ValeriiSudakov**
- Renamed TaskDTO to TaskServiceDTO
- Added TaskRepositoryDTO.
- Made TaskRepository the isolated, independent module. 
- Now TaskEntity is internal class of TaskRepository. Input and output task data is TaskRepositoryDTO

### CODE REVIEW

File: core/API/TaskServiceDTO.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+const std::string TaskServiceDTO::ToString() const {
+  std::string result;
+  std::string priority = PriorityToString();
+  result += "Task name: " + name_ + "\nTask label: " + label_ + "\nTask priority: " + priority
+          + (taskComplete_ ? "\nCompleted" : "\nNot completed") + "\nDate: " + date_.ToString();
+  return result;
+}
+
```
Comment (RodionShyshkin) [inline core/API/TaskServiceDTO.cpp:69]: Do you really need this method?

Comment (ValeriiSudakov) [reply core/API/TaskServiceDTO.cpp:69]: Yeap
Its using in CLI, or wherever else.

Comment (RodionShyshkin) [inline core/API/TaskServiceDTO.cpp:69]: Do you really need this method?

File: core/Memory_Model/Storage/TaskStorage.h [hunk 2]
```diff
@@ -14,8 +14,7 @@ class TaskStorage{
  public:
   virtual const std::map<TaskID, std::shared_ptr<TaskEntity>>&    GetTasks() const = 0;
   virtual std::optional<std::shared_ptr<TaskEntity>>              GetTask(const TaskID& taskID) const = 0;
-  virtual std::optional<std::shared_ptr<TaskEntity>>              AddTask(const Task& task) = 0;
-  virtual std::optional<std::shared_ptr<TaskEntity>>              AddSubtask(const TaskID& rootTaskID, const Task& subtask) = 0;
+  virtual std::optional<std::shared_ptr<TaskEntity>>              AddTask(const TaskEntity& task) = 0;
```
Comment (RodionShyshkin) [inline core/Memory_Model/Storage/TaskStorage.h:17]: For what have you changed Task for TaskEntity?

File: core/Memory_Model/Task/TaskEntity.h [hunk 3]
```diff
@@ -34,11 +34,11 @@ class TaskEntity {
...
-  TaskID                                                id_;
-  TaskID                                                parentID_;
-  bool                                                  complete_;
-  std::map<TaskID, std::weak_ptr<TaskEntity>>           subtasks_;
+  Task                                                      task_;
+  TaskID                                                    id_;
+  TaskID                                                    parentID_;
+  bool                                                      complete_;
+  std::map<TaskID, std::weak_ptr<TaskEntity>>               subtasks_;
 };
```
Comment (RodionShyshkin) [inline core/Memory_Model/Task/TaskEntity.h:42]: Refactoring should be in separate PR, you're annigilyatornaya pushka

**Review by RodionShyshkin — APPROVED** [review]

File: core/Memory_Model/Task/TaskEntity.h (hunk 3 above)
Comment (ValeriiSudakov) [reply core/Memory_Model/Task/TaskEntity.h:42]: oh, shit, i'm sorry ahhahah

File: core/Memory_Model/Storage/TaskStorage.h (hunk 2 above)
Comment (ValeriiSudakov) [reply core/Memory_Model/Storage/TaskStorage.h:17]: because for now, TaskEntity are creating in TaskRepository.

Comment (ValeriiSudakov) [reply core/Memory_Model/Storage/TaskStorage.h:17]: Task was an input parameter, because ID for taskEntity was generated in taskStorage. And now it is in taskRepository.

**Review by RodionShyshkin — APPROVED** [review]
